editors/vscode/client/extension.ts

```diff
@@ -251,3 +277,34 @@ export function deactivate(): Thenable<void> | undefined {
   }
   return client.stop();
 }
+
+function createFileEventWatchers(configRelativePath: string) {
+  const workspaceConfigPatterns = (workspace.workspaceFolders || []).map((workspaceFolder) =>
+    new RelativePattern(workspaceFolder, configRelativePath)
+  );
+
+  return [
+    workspace.createFileSystemWatcher(`**/${oxlintConfigFileName}`),
+    ...workspaceConfigPatterns.map((pattern) => {
+      return workspace.createFileSystemWatcher(pattern);
+    }),
+  ];
+}
+
+async function findAllConfigFiles(configRelativePath: string) {
```

Comment:
is this really needed? 
I expect `workspace.createFileSystemWatcher("**/${oxlintConfigFileName}")` to find all configs and send the `workspace/didChangeWatchedFiles` to the server

Reply:
It's only needed with a user supplied path, such as `oxlint.json` or whatever it may be that isn't using the standard `.oxlintrc.json`.

Reply:
`oxlint` does not support custom filenames for nested configuration, see PR https://github.com/oxc-project/oxc/pull/9153

Reply:
With how this is used, I don't think we need to find the user supplied config files.

We do still need to be able to find the `.oxlintrc.json` files and send those over on language server startup. `workspace.createFileSystemWatcher` only watches for events, it doesn't trigger a create event on startup for files that were already there before the watcher was registered.

Reply:
Updated to only find the one config file name. Resolving this, but please re-open if it needs more discussion.